### PR TITLE
Update, review, and rename K statistic

### PR DESCRIPTION
**Pull request by jGaboardi**
This PR follows up #465, #470, #471, etc. to address #469.

See also #450.

### CODE REVIEW

**Comment by jGaboardi** [comment]
test out @ljwolf's example:
```python
import numpy
from scipy.spatial import distance 

coordinates = numpy.random.uniform(size=(1000,2))
distances = distance.pdist(coordinates) # this is flat, not square!
support_for_kfunction = numpy.linspace(0, distances.max(), num=100).reshape(-1,1)

(distances < support_for_kfunction).sum(axis=1)
```
> if you just look at that `(distances < support_for_kfunction)`, it's a 1000 x 100 matrix (or, *n* observations by *k* bins) matrix where *M[i,j]* is `True` when distance *i* is less than the bin distance *j*

**Comment by jGaboardi** [comment]
Vectorization schema implemented in 4f40f16. See [this gist](https://gist.github.com/jGaboardi/3dbf19363d68b91e7c9764cf358519d2) for testing.

TL;DR — According to the tests in the gist we can expect 2x-4x speedups.
